### PR TITLE
mac: add a getter function for the MAC algorithms' block size

### LINKED ISSUE #12342: Add EVP_MAC_CTX getter for block size
While porting CPython to the new MAC API I noticed that EVP_MAC_CTX is missing a getter to retrieve the block size of MAC for HMAC. The block size is part of Python's HMAC API. For OpenSSL 1.1.1 and earlier CPython uses

```
    const EVP_MD *md = HMAC_CTX_get_md(self->ctx);
    return PyLong_FromLong(EVP_MD_block_size(md));
```

to get the block size.

### PR DESCRIPTION
**Pull request by paulidale**
- [x] documentation is added or updated
- [x] tests are added or updated

Fixes #12342

### CODE REVIEW

**Comment by levitte** [comment]
It's a mystery to me why the block size would be interesting to any code outside of the algorithm itself, so I'm not sure what to do with this PR.

File: doc/man3/EVP_MAC.pod [hunk 1]
```diff
@@ -381,6 +385,9 @@ return 1 on success, 0 on error.
 EVP_MAC_CTX_get_mac_size() returns the expected output size, or 0 if it isn't set.
 If it isn't set, a call to EVP_MAC_init() should get it set.
 
+EVP_MAC_CTX_get_block_size() returns the block size, or 0 if it isn't set.
+If it isn't set, a call to EVP_MAC_init() should get it set.
```
Comment (slontis) [inline doc/man3/EVP_MAC.pod:389]: get it set?

Comment (paulidale) [reply doc/man3/EVP_MAC.pod:389]: That's what was written for three lines up which I copied from :)

**Comment by paulidale** [comment]
> It's a mystery to me why the block size would be interesting to any code outside of the algorithm itself, so I'm not sure what to do with this PR.

Ask the person who raised the issue perhaps?

**Comment by levitte** [comment]
Ah, [RFC2104](https://datatracker.ietf.org/doc/html/rfc2104#section-2) to the rescue.  The key length (K) is limited by the hash output length (L) and the digest block size (B): L \<= K \<= B
So if the application wants to be able to check the lengths for themselves, they need the block size.

Ok then...

File: util/libcrypto.num [hunk 2]
```diff
@@ -5413,4 +5413,5 @@ BIO_get_line                            5540	3_0_0	EXIST::FUNCTION:
 OSSL_LIB_CTX_new_from_dispatch          5541	3_0_0	EXIST::FUNCTION:
 OSSL_LIB_CTX_new_child                  5542	3_0_0	EXIST::FUNCTION:
 OSSL_PROVIDER_get0_dispatch             5543	3_0_0	EXIST::FUNCTION:
+EVP_MAC_CTX_get_block_size              5544	3_0_0	EXIST::FUNCTION:
 PKCS5_PBE_keyivgen_ex                   5544	3_0_0	EXIST::FUNCTION:
```
Comment (levitte) [inline util/libcrypto.num:5417]: Undo that and to `make update` again

Comment (paulidale) [reply util/libcrypto.num:5416]: Done.

**Review by levitte — APPROVED** [review]

**Review by slontis — COMMENTED** [review]
Update provider-mac also?

**Comment by paulidale** [comment]
Updated.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master.  Thanks for the feedback.

**Pull request closed without merging**